### PR TITLE
refactor: move Boost Datetime usage to wallet

### PR DESCRIPTION
**Pull request by fanquake**
This means we don't need Boost Datetime in a `--disable-wallet` build, and it isn't included in the kernel (via time.h/cpp). Split from a larger boost removal branch/effort.

### CODE REVIEW

**Review by aureleoules — APPROVED** [review]
ACK aec34489854198908424b6ef61142315533fd074

**Comment by hebasto** [comment]
> This means we don't need Boost Datetime in a `--disable-wallet` build,

Could it be enforced to avoid reintroducing by accident in the future? `lint-includes.py`?

File: src/test/fuzz/util.cpp [hunk 1]
```diff
@@ -307,8 +307,8 @@ CAmount ConsumeMoney(FuzzedDataProvider& fuzzed_data_provider, const std::option
 int64_t ConsumeTime(FuzzedDataProvider& fuzzed_data_provider, const std::optional<int64_t>& min, const std::optional<int64_t>& max) noexcept
 {
     // Avoid t=0 (1970-01-01T00:00:00Z) since SetMockTime(0) disables mocktime.
-    static const int64_t time_min{ParseISO8601DateTime("2000-01-01T00:00:01Z")};
-    static const int64_t time_max{ParseISO8601DateTime("2100-12-31T23:59:59Z")};
+    static const int64_t time_min{946684801}; // 2000-01-01T00:00:01Z
+    static const int64_t time_max{4133980799}; // 2100-12-31T23:59:59Z
```
Comment (hebasto) [inline src/test/fuzz/util.cpp:311]: nit: maybe add
```
    BOOST_CHECK_EQUAL(ParseISO8601DateTime("2000-01-01T00:00:01Z"), 946684801);
    BOOST_CHECK_EQUAL(ParseISO8601DateTime("2100-12-31T23:59:59Z"), 4133980799);
```
to `src/wallet/test/rpc_util_tests.cpp`?

**Review by hebasto — APPROVED** [review]
ACK aec34489854198908424b6ef61142315533fd074, I have reviewed the code and it looks OK, I agree it can be merged.

**Comment by theuni** [comment]
Concept ACK.

**Comment by theuni** [comment]
ACK aec34489854198908424b6ef61142315533fd074. Agree with @hebasto's comment to avoid losing some test coverage.

Nice to relegate this to wallet only.

File: src/test/fuzz/util.cpp (hunk 1 above)
Comment (JohhnyBTC) [reply src/test/fuzz/util.cpp:311]: 🚣

Comment (fanquake) [reply src/test/fuzz/util.cpp:311]: Done

**Review by hebasto — APPROVED** [review]
re-ACK 079cf88c0df6de038b8f8933d55c0d17af007b43

**Review by aureleoules — APPROVED** [review]
re-ACK 079cf88c0df6de038b8f8933d55c0d17af007b43 - rebased and two additional unit tests since my last review.

**Review by jarolrod — COMMENTED** [review]
crACK 079cf88c0df6de038b8f8933d55c0d17af007b43

**Comment by MarcusMWilliams** [comment]
Marcus Maurice Williams

**Comment by MarcoFalke** [comment]
This is reverting 9e2c623be50ee7e586a411923b9ed136acfa2b3f, but seems fine if the goal is to remove boost from kernel?

**Comment by fanquake** [comment]
> but seems fine if the goal is to remove boost from kernel?

Yes, and removing it from bitcoind+utils more generally. There's no reason for this wallet-only Boost usage to be sucked into everything else.

**Comment by MarcoFalke** [comment]
Right, I can't see a use case to parse this outside of the wallet, so lgtm

**Pull request merged**